Move client to stable (#93)
status-im/specs: status-client-spec.md

            -   [What about other RLPx subprotocols like LES, and Swarm?](#what-about-other-rlpx-subprotocols-like-les-and-swarm)
            -   [Why do you use Whisper?](#why-do-you-use-whisper)
            -   [I heard you were moving away from Whisper?](#i-heard-you-were-moving-away-from-whisper)
            -   [Why is PoW for Whisper set so low?](#why-is-pow-for-whisper-set-so-low)
            -   [Why do you not use Discovery v5 for node discovery?](#why-do-you-not-use-discovery-v5-for-node-discovery)
            -   [I heard something about mailservers being trusted somehow?](#i-heard-something-about-mailservers-being-trusted-somehow)
        -   [Data sync](#data-sync-1)
            -   [Why is MVDS not used for public chats?](#why-is-mvds-not-used-for-public-chats)
    -   [Footnotes](#footnotes)
    -   [Appendix A: Security considerations](#appendix-a-security-considerations)
        -   [Scalability and UX](#scalability-and-ux)
        -   [Privacy](#privacy)
        -   [Spam resistance](#spam-resistance)
        -   [Censorship resistance](#censorship-resistance)
    -   [Acknowledgements](#acknowledgements)

## Introduction

> The Status mobile app is a multifunction application. This document mainly describes the requirements for establishing connections and communicating between clients. The other features specific to the Status application can be found as follows:

Author: Sorry I don't like this section, it links to no-op work and the story it tells is weird. Document deals with way more than connections, and the main thing that are specified aren't liked. We can reword this into PR once other specs are further along. Removing.

        -   [Censorship resistance](#censorship-resistance)
    -   [Acknowledgements](#acknowledgements)

### Protocol layers

Implementing a Status clients largely means implementing the following layers. Additionally, there are separate specifications for things like key management and account lifecycle.

> Some things, such as storage and blockchain usage, are currently underspecified.

Author: ```suggestion
Other aspects, such as how IPFS is used for stickers, how we interact with the Ethereum blockchain or how the browser works, are currently underspecified. These sets of specifications should allow you to implement a a Status client for basic private communication.
```

Author: cc @corpetty fyi

### P2P Overlay

> Status clients run on a private peer-to-peer network, as specified by the devP2P

Author: ```suggestion
Status clients run on a public, permissionless peer-to-peer network, as specified by the devP2P
```

### P2P Overlay

> Status clients run on a private peer-to-peer network, as specified by the devP2P

Author: Is this accurate? It is permissionless, nothing is stopping anyone from running a node. @corpetty

Reviewer: good point, maybe `separate, permissionless` instead of `private`?

Reviewer: or just `permissionless`

Author: I'd say public and permissionless

Currently the main bootstrap nodes are provided by Status Gmbh, but anyone can
run these provided they are connected to the rest of the Whisper network.

Status maintains a list of production fleet boootstrap nodes in the following locations:
- Hong Kong: 

>    - "enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443"

Reviewer 2: Maybe use backticks instead of double quotes for those?

Author: done

>    - "enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443"

Author: ```suggestion
   - `enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443`
```

**Hong Kong:**
   - `enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443`
   - `enode://23d0740b11919358625d79d4cac7d50a34d79e9c69e16831c5c70573757a1f5d7d884510bc595d7ee4da3c1508adf87bbc9e9260d804ef03f8c1e37f2fb2fc69@47.52.106.107:443`
**Amsterdam:**
   - `enode://436cc6f674928fdc9a9f7990f2944002b685d1c37f025c1be425185b5b1f0900feaf1ccc2a6130268f9901be4a7d252f37302c8335a2c1a62736e9232691cc3a@178.128.138.128:443`
    - `enode://5395aab7833f1ecb671b59bf0521cf20224fe8162fc3d2675de4ee4d5636a75ec32d13268fc184df8d1ddfa803943906882da62a4df42d4fccf6d17808156a87@178.128.140.188:443`
**Central US:**
   - `enode://32ff6d88760b0947a3dee54ceff4d8d7f0b4c023c6dad34568615fcae89e26cc2753f28f12485a4116c977be937a72665116596265aa0736b53d46b27446296a@34.70.75.208:443`
   - `enode://5405c509df683c962e7c9470b251bb679dd6978f82d5b469f1f6c64d11d50fbd5dd9f7801c6ad51f3b20a5f6c7ffe248cc9ab223f8bcbaeaf14bb1c0ef295fd0@35.223.215.156:443`

These bootstrap nodes MAY change and we can't guarantee that it will stay this way forever
and at some point we might be forced to change them.

#### Discovery

> To implement a Status client you MUST discover peers to connect to. We use a

Author: ```suggestion
To implement a Status client you MUST discover or have a list of peers to connect to. We use a
```

3. Mailservers (servers and clients)
4. Mobile nodes (Status Clients)

> To implement a standard Status client you MUST implement both 2. and 4. node types. The

Author: Why are we adding 2 ?

Status maintains a list of production fleet boootstrap nodes in the following locations:

> - Hong Kong: 

Reviewer 3: ```suggestion
**Hong Kong:**
```

   - `enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443`
   - `enode://23d0740b11919358625d79d4cac7d50a34d79e9c69e16831c5c70573757a1f5d7d884510bc595d7ee4da3c1508adf87bbc9e9260d804ef03f8c1e37f2fb2fc69@47.52.106.107:443`

> - Amsterdam: 

Reviewer 3: ```suggestion
**Amsterdam:**
```

   - `enode://436cc6f674928fdc9a9f7990f2944002b685d1c37f025c1be425185b5b1f0900feaf1ccc2a6130268f9901be4a7d252f37302c8335a2c1a62736e9232691cc3a@178.128.138.128:443`
    - `enode://5395aab7833f1ecb671b59bf0521cf20224fe8162fc3d2675de4ee4d5636a75ec32d13268fc184df8d1ddfa803943906882da62a4df42d4fccf6d17808156a87@178.128.140.188:443`

> - Central US: 

Reviewer 3: ```suggestion
**Central US:**
```

#### Discovery

> To implement a Status client you MUST discover peers to connect to. We use a

Author: Is this accurate? If you have nodes ready as static nodes you don't necessarily need to discover new nodes. SHOULD?

cc @adambabik

Reviewer: have a list of static nodes is trivial discovery, imo

with the `whisper` topic. Status nodes that are mail servers and want to
be discoverable MUST additionally register with the `whispermail` topic.

> It is RECOMMENDED to use both mechanisms but at the same time implement a structure

Author: Is this an official term as per RFC?

Reviewer: > 3. SHOULD   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.

## Security Considerations

> See [Appendix A](#appendix-a-security-considerations)

Author: This link seems weird?

Reviewer: why? it just goes to the appendix at the bottom of the page

Author: ah, my bad, this is middle
